src/Orleans.Runtime/Timers/AsyncTimer.cs

```diff
@@ -84,6 +86,16 @@ public bool CheckHealth(DateTime lastCheckTime)
             return true;
         }
 
-        public void Dispose() => this.cancellation?.Cancel(throwOnFirstException: false);
+        public void Cancel() => this.cancellation.Cancel(throwOnFirstException: false);
+
+        public void Dispose()
+        {
+            if (!this.disposed)
+            {
+                this.disposed = true;
+                this.Cancel();
+                this.cancellation.Dispose();
+            }
```

Comment:
We have a 'disposed' check here because calling `CancellationTokenSource.Cancel()` might throw if the CTS is already disposed